web/src/popover_menus.ts

```diff
@@ -411,11 +416,22 @@ export function toggle_popover_menu(
 
     // If the window is mobile-sized, we will render the
     // popover centered on the screen as an overlay.
-    if (
+    let show_as_overlay =
         (options?.show_as_overlay_on_mobile === true &&
             window.innerWidth <= media_breakpoints_num.md) ||
-        options?.show_as_overlay_always === true
-    ) {
+        options?.show_as_overlay_always === true;
+    // Show the popover as over if the reference element is hidden.
+    if (!show_as_overlay && options?.show_as_overlay_if_reference_hidden_at_trigger) {
+        const target_props = $(target).get_offset_to_window();
+        const viewport_info = message_viewport.message_viewport_info();
+        if (
+            target_props.top < viewport_info.visible_top ||
+            target_props.bottom > viewport_info.visible_bottom
+        ) {
+            show_as_overlay = true;
+        }
```

Comment:
I was going to suggest we might want to think about making all popovers do the overlay thing if their reference is offscreen.

But I think this logic at least will only work for the message feed, given `message_viewport_info`.

Reply:
Yeah